libr/search/search.c

```diff
@@ -439,11 +360,11 @@ R_API void r_search_kw_reset(RSearch *s) {
 }
 
 R_API void r_search_reset(RSearch *s, int mode) {
+	R_FREE (s->data);
```

Comment:
This usages is like calling a member function where the first argument denotes the instance. The call site should make this check and it would be too defensive to add such checks.